### PR TITLE
#5 Allow tasks to be disabled

### PR DESCRIPTION
**Pull request by rdwallis**

### CODE REVIEW

**Comment by rdwallis** [comment]
@omega @yarcub I don't have any native apps to test on can you check if this works:

You need to turn autoCreate to false and then in your bootstrapper or entry point call something like:

```
analytics.create().clientId(UNIQUE_ID_SINCE_COOKIES_ARENT_AVAILABLE).go();
analytics.setGlobalSettings().disableTask(Task.CHECK_PROTOCOL).go();
analytics.sendPageView().go();
```

@yarcub you can do your noop stuff by also disabling `Task.SEND_HIT`, For logging you can use the existing goAndLog method. I'll work out something better when I have some time.

Completely untested so please report back if it works or doesn't.

**Comment by omega** [comment]
Did you mean to tag someone else than me perhaps? If you did indeed intend to tag me, I'm sorry, but I must have forgotten what this is about, so a light reminder would be great :)

**Comment by rdwallis** [comment]
Ah yes sorry meant to tag @0mega

File: src/main/java/com/arcbees/analytics/shared/Task.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2014 ArcBees Inc.
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/analytics/shared/Task.java]: 2015

**Comment by christiangoudreau** [comment]
Looks good, but can we have some tests?

**Comment by rdwallis** [comment]
Added issue #41.  We need tests.  I don't have time to write them so someone else will need to volunteer.

This pr needs feedback from someone building for native before I'll merge.  Since I haven't checked if this works at all someone else will need to confirm.

**Comment by 0mega** [comment]
@rdwallis Thanks for fast feedback, I'm still working on this. There need to be another option in place for this to work on mobile (disable cookies). As soon as I finish testing I'll make a PR.

Regarding test I'll try to see a way of at least starting with some simple cases, but no promises.

**Comment by rdwallis** [comment]
Ok yes I see there's a storage: 'none' feature mentioned in a couple of places but it isn't documented at https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference

I'll add it in the next few days.  Would be nice to know what values you can pass to it other than `none`

**Comment by rdwallis** [comment]
Ok I've added the ability to set storage.  And a storage enum with only one value so we can expand it when we work out what the possible values are.

@0mega you should be able to set this up:
```
analytics.create().clientId(UNIQUE_ID_SINCE_COOKIES_ARENT_AVAILABLE).storage(Storage.NONE).go();
analytics.setGlobalSettings().disableTask(Task.CHECK_PROTOCOL).go();
analytics.sendPageView().go();
```

Let me know if it works.

**Comment by 0mega** [comment]
Hi @rdwallis, regarding to the code, it's working fine :+1:, I just added two convenience methods for sending [ScreenView](https://developers.google.com/analytics/devguides/collection/analyticsjs/screens) from mobile instead of PageView in this PR #42. So everything is working in native apps.


I guess the `storage` has only one value `none` and it's only purpose is to disable the cookies (not 100% sure). So we could have a method `CreateOptions#disableCookies()` instead of `CreateOptions#storage(Storage)`.

Still didn't have time to start writing some tests :disappointed:.

**Comment by rdwallis** [comment]
The storage enum is stupid now but it's more future proof in case google adds other storage options. It's not a boolean in analytics.js which points to it having more than one value in the future.

**Comment by rdwallis** [comment]
@christiangoudreau @Chris-V just waiting for a lgtm and I'll merge this.

**Comment by christiangoudreau** [comment]
You have mine, but wait for Chris-V :D

**Comment by Chris-V** [comment]
LGTM

**Pull request merged**